aws/resource_aws_backup_selection.go

```diff
@@ -0,0 +1,182 @@
+)
+
+func resourceAwsBackupSelection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBackupSelectionCreate,
+		Read:   resourceAwsBackupSelectionRead,
+		Delete: resourceAwsBackupSelectionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"plan_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"iam_role": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tag": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAwsConditionTagHash,
+			},
+			"resources": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceAwsBackupSelectionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).backupconn
+
+	selection := &backup.Selection{}
+
+	selection.SelectionName = aws.String(d.Get("name").(string))
+	selection.IamRoleArn = aws.String(d.Get("iam_role").(string))
+	selection.ListOfTags = gatherConditionTags(d)
+	selection.Resources = expandStringList(d.Get("resources").([]interface{}))
+
+	input := &backup.CreateBackupSelectionInput{
+		BackupPlanId:    aws.String(d.Get("plan_id").(string)),
+		BackupSelection: selection,
+	}
+
+	resp, err := conn.CreateBackupSelection(input)
+	if err != nil {
+		return fmt.Errorf("error creating Backup Selection: %s", err)
+	}
+
+	d.SetId(*resp.SelectionId)
+
+	return resourceAwsBackupSelectionRead(d, meta)
+}
+
+func resourceAwsBackupSelectionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).backupconn
+
+	input := &backup.GetBackupSelectionInput{
+		BackupPlanId: aws.String(d.Get("plan_id").(string)),
+		SelectionId:  aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetBackupSelection(input)
+	if err != nil {
+		return fmt.Errorf("error reading Backup Selection: %s", err)
+	}
+
+	d.Set("plan_id", resp.BackupPlanId)
+	d.Set("name", resp.BackupSelection.SelectionName)
+	d.Set("iam_role", resp.BackupSelection.IamRoleArn)
+
+	if resp.BackupSelection.ListOfTags != nil {
+		tag := &schema.Set{F: resourceAwsConditionTagHash}
+
+		for _, r := range resp.BackupSelection.ListOfTags {
+			m := make(map[string]interface{})
+
+			m["type"] = *r.ConditionType
+			m["key"] = *r.ConditionKey
+			m["value"] = *r.ConditionValue
+
+			tag.Add(m)
+		}
+
+		d.Set("tag", tag)
+	}
+	if resp.BackupSelection.Resources != nil {
+		d.Set("resources", resp.BackupSelection.Resources)
```

Comment:
Same here with regards to error checking:

```suggestion
		if err := d.Set("resources", resp.BackupSelection.Resources); err != nil {
			return fmt.Errorf("error setting resources: %s", err)
		}
```

I believe in this case you may see that it needs `aws.StringValueSlice()` or `flattenStringList()` called on it first.